### DISCUSSION TITLE
Upgrade to newest git commit fails using docker-compose

### DISCUSSION THREAD

**Discussion by bokkabonga** [dani-garcia/vaultwarden#2602] [Q&A]
### Subject of the issue
When trying to upgrade from 1.24 to the newest version using "docker-compose build" it fails to build.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.24.0-08f0de7b
* Web-vault version: v2.25.1
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 10.4.20-MariaDB-1:10.4.20+maria~focal
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SENDS_ALLOWED, HIBP_API_KEY, USER_ATTACHMENT_LIMIT, ORG_ATTACHMENT_LIMIT, TRASH_AUTO_DELETE_DAYS, SIGNUPS_ALLOWED, SIGNUPS_VERIFY, SIGNUPS_VERIFY_RESEND_TIME, SIGNUPS_VERIFY_RESEND_LIMIT, SIGNUPS_DOMAINS_WHITELIST, INVITATIONS_ALLOWED, ADMIN_TOKEN, INVITATION_ORG_NAME, DISABLE_2FA_REMEMBER, YUBICO_CLIENT_ID, YUBICO_SECRET_KEY, SMTP_HOST, SMTP_SSL, SMTP_PORT, SMTP_FROM, EMAIL_TOKEN_SIZE, EMAIL_EXPIRATION_TIME

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "*****://****:*********************************@**/*********",
  "db_connection_retries": 10,
  "disable_2fa_remember": true,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*********.***.**",
  "domain_origin": "*****://*********.***.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": false,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": "***",
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Bitwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/bitwarden.log",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": 20480,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "***.**",
  "signups_verify": true,
  "signups_verify_resend_limit": 3,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*********@***.**",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": "**********.***.**",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "trash_auto_delete_days": 60,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": 5121,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": "61202",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

* Install method: docker-compose

* MySQL/MariaDB or PostgreSQL version: MariaDB
* OS: Oracle 7.9

### Steps to reproduce

1. Pull latest repository using git pull.
2. Stop and prune old container.
3. Try to build new container using "docker-compose build"

### Expected behaviour
Building the container passes without problem

### Actual behaviour
Building fails with Error (see below)

### Troubleshooting data
" => ERROR [build 12/13] RUN cargo build --features mysql --release        37.8s
------
 > [build 12/13] RUN cargo build --features mysql --release:
#0 1.195    Compiling vaultwarden v1.0.0 (/app)
#0 37.79 error: could not compile `vaultwarden`
#0 37.79 
#0 37.79 Caused by:
#0 37.79   process didn't exit successfully: `rustc --crate-name vaultwarden --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto=thin --cfg 'feature="mysql"' -C metadata=335c0506652fb368 -C extra-filename=-335c0506652fb368 --out-dir /app/target/release/deps -C strip=debuginfo -L dependency=/app/target/release/deps --extern backtrace=/app/target/release/deps/libbacktrace-ca1b17b49b3d2570.rlib --extern bytes=/app/target/release/deps/libbytes-28a55cf7a9873cad.rlib --extern cached=/app/target/release/deps/libcached-b06ebce73f2df54e.rlib --extern chrono=/app/target/release/deps/libchrono-46217a44df528449.rlib --extern chrono_tz=/app/target/release/deps/libchrono_tz-0d6a3e1d82478237.rlib --extern cookie=/app/target/release/deps/libcookie-c8e9e219d7e217e9.rlib --extern cookie_store=/app/target/release/deps/libcookie_store-1dcf50ff1fe194d4.rlib --extern ctrlc=/app/target/release/deps/libctrlc-22be4af2a872a5d8.rlib --extern dashmap=/app/target/release/deps/libdashmap-a00bc501fcb651be.rlib --extern data_encoding=/app/target/release/deps/libdata_encoding-0df089d7a552f827.rlib --extern data_url=/app/target/release/deps/libdata_url-b6739ab45954d249.rlib --extern diesel=/app/target/release/deps/libdiesel-7e6ed18516f5ec0d.rlib --extern diesel_migrations=/app/target/release/deps/libdiesel_migrations-892cb6fe254deff7.rlib --extern dotenvy=/app/target/release/deps/libdotenvy-59114528165f918e.rlib --extern fern=/app/target/release/deps/libfern-8ff983cc66174072.rlib --extern futures=/app/target/release/deps/libfutures-5c5daa0c0bc46e82.rlib --extern governor=/app/target/release/deps/libgovernor-17dd9f111daf58f4.rlib --extern handlebars=/app/target/release/deps/libhandlebars-aea0201ad561fe79.rlib --extern html5gum=/app/target/release/deps/libhtml5gum-6d0366c1dc3e4ebd.rlib --extern job_scheduler_ng=/app/target/release/deps/libjob_scheduler_ng-9968acae665688ae.rlib --extern jsonwebtoken=/app/target/release/deps/libjsonwebtoken-896e78f5c3ad97da.rlib --extern lettre=/app/target/release/deps/liblettre-2750355cffc7d99e.rlib --extern log=/app/target/release/deps/liblog-79690080283570e5.rlib --extern num_derive=/app/target/release/deps/libnum_derive-481258659e54557d.so --extern num_traits=/app/target/release/deps/libnum_traits-d0a073c728b817f1.rlib --extern once_cell=/app/target/release/deps/libonce_cell-e8ce1beeee82a17c.rlib --extern openssl=/app/target/release/deps/libopenssl-183713e0dffae895.rlib --extern paste=/app/target/release/deps/libpaste-fd90a7e4edc02652.so --extern percent_encoding=/app/target/release/deps/libpercent_encoding-39a45f3b6cbe8393.rlib --extern pico_args=/app/target/release/deps/libpico_args-25e1e42db6acf6c2.rlib --extern rand=/app/target/release/deps/librand-75fe3dffc24902eb.rlib --extern regex=/app/target/release/deps/libregex-8ee2a383b6891f19.rlib --extern reqwest=/app/target/release/deps/libreqwest-28d97eba5c29edf6.rlib --extern ring=/app/target/release/deps/libring-81ac9b134372da6e.rlib --extern rmpv=/app/target/release/deps/librmpv-a36de11838723094.rlib --extern rocket=/app/target/release/deps/librocket-4fecf82a8cdcfcc6.rlib --extern serde=/app/target/release/deps/libserde-b34e1a525c2323d2.rlib --extern serde_json=/app/target/release/deps/libserde_json-f5296bfa96a2e7ee.rlib --extern syslog=/app/target/release/deps/libsyslog-c85a42c6b05d89f3.rlib --extern time=/app/target/release/deps/libtime-9c19f8a0b8d00534.rlib --extern tokio=/app/target/release/deps/libtokio-599514154ee33497.rlib --extern tokio_tungstenite=/app/target/release/deps/libtokio_tungstenite-afcf4032eafdda8c.rlib --extern totp_lite=/app/target/release/deps/libtotp_lite-163af3eb92d900a5.rlib --extern tracing=/app/target/release/deps/libtracing-04b7b404960f44ba.rlib --extern url=/app/target/release/deps/liburl-668390734e1ab01b.rlib --extern uuid=/app/target/release/deps/libuuid-47230f0f238dc641.rlib --extern webauthn_rs=/app/target/release/deps/libwebauthn_rs-e82d8fffff6e3ea4.rlib --extern yubico=/app/target/release/deps/libyubico-48ddbdc6eb9e9d38.rlib --cfg mysql -L native=/usr/lib/x86_64-linux-gnu/ -L native=/app/target/release/build/ring-16c4279a66673faa/out` (signal: 9, SIGKILL: kill)
------
failed to solve: process "/bin/sh -c cargo build --features ${DB} --release" did not complete successfully: exit code: 101"

**Comment by bokkabonga** [comment]
The same procedure works fine on Oracle 8.5. So i suspect it has something to do with some dependency maybe?

**Comment by BlackDex** [comment]
Well, dependencies would be strange as those should all be inside the container. A container build can't be different on two separate hosts with the exact same git checksum.

I see a `sigkill kill` there, that does look like depleted resources like memory for example.

**Comment by bokkabonga** [comment]
Yeah you are completely right. The server only had 2 GB of RAM. It seems thats not enough anymore. I assigned 4 GB to the server now and it works again like a charm :)
